packages/remix-server-runtime/data.ts

```diff
@@ -104,7 +135,38 @@ export async function callRouteLoader({
     );
   }
 
-  return isResponse(result) ? result : json(result);
```

Comment:
I think it would be better to create a `warnIfNotResponse()` function and use it in both places.